### ISSUE TITLE
[Question] Is this project abandoned?

### ISSUE THREAD

**Issue by gracie95** [dslul/openboard#787]

**Comment by luxcalor** [comment]
I wonder too

**Comment by philibe** [comment]
IMHO 6 months without commit is a too small duration to conclude that.

**Comment by Helium314** [comment]
> IMHO 6 months without commit is a too small duration to conclude that.

It's not really the lack of commits, rather the lack of interaction with the devs at all.
No one knows whether the devs might come back, or maybe hand the project to some more active maintainers, or whether they're not interested in doing any change for openboard at all.
If it's clear the latter is the case, then one could simply create a fork, implement some of the features sitting around in PRs, and put it on F-Droid... and then hopefully continue maintaining it!

**Comment by bushrang3r** [comment]
Would be great if someone would fork and continue the app.

**Comment by Ansarmughal99** [comment]
This project should not be abandoned .it is a great project .

**Comment by l3u** [comment]
It would be great to see some action here again (or in a fork, if the project is actually dead). We don't have too much alternatives for a free keyboard :-(

**Comment by alekksander** [comment]
also hoping either for some new commits or foss alternatvie to this one, as for now it's (still) unbeatable

**Comment by supernatural30** [comment]
I'm eager for a new apk too. Most feature rich FOSS keyboards seem to be stagnant right now.

**Comment by ghost** [comment]
I'm looking forward to a fork, too. It's the best FOSS keyboard out there. FlorisBoard is way too unfinished, and I dislike AnySoft keyboard. These three keyboards are the only decent options. Really hope someone will take on this project. @dslul seems to be relatively active. Maybe someone could contact him regarding the project.

**Comment by alekksander** [comment]
>These three keyboards are the only decent options. Really hope someone will take on this project. 

there are at least two more good, working projects still in active development, but they lack features (both named simple keyboard)

**Comment by ghost** [comment]
> there are at least two more good, working projects still in active development, but they lack features (both named simple keyboard)

I know them and tried them. The reason I didn't list them is precisely because they are not decent for daily usage at all. They are only good for some really simple input, maybe for some specific use-cases, but not for daily usage. Ideally, there should be an open-source clone of Google Keyboard and SwiftKey.

The current state of open-source Android keyboards is apocalyptic. I tested all the keyboards available, both proprietary and open-source, and it turned out to be the best option to use SwiftKey (I use SwiftKey Beta because I like to receive new stuff early and it's no less stable than regular SwiftKey) with internet disabled. I consider SwiftKey as the best Android keyboard available. Yes, it has 3 Google analytics trackers and its own Microsoft telemetry, and has a `com.google.android.gms.permission.AD_ID` listed in permissions, but these can be mitigated by firewalling the app. All other keyboards are simply not up to par and make me much less productive. I know there are people who flat out refuse to use proprietary keyboards (or software in general). They cannot even: they don't have a Google account, can't even login to Play Store and download SwiftKey or Gboard (but it is possible to download an .apk from Play Store via a specific website, though it's a less safe method than Play Store. And they will have to download a new .apk manually when there's an update). In my opinion, this is a deliberate crippling of oneself. Google is definitely toxic, but the world shouldn't be all black or white, and to be as efficient as possible, we have to make some trade-offs between privacy/security and being productive.

**Comment by getchup** [comment]
I'm gonna continue using this, even if it's abandoned.

**Comment by philibe** [comment]
TLDR: same as @getchup 


I've found and choose OpenBoard for 3 majors goals:

- No tracking, no account, no cloud
- to be able to deactivate all pre-input, AI and so on: a simple keyboard, without useless gadgets. 
- Open source

The only keyboard which I have used and liked very much because it was powerful and simple, with the choice to not make coffee with, was Kii Keyboard.

I cannot work with SwiftKey: all  the time it is going to propose something out of subject, why it proposes all the time my town or my name ? Their AI is a little stupid. And not much better with Google Keyboard.

Same as @getchup : I'm gonna continue using this,  even if it's abandoned, or a fork if it will dead, if possible never an other keyboard which want to make cakes or IA (with tracking)

**Comment by ghost** [comment]
> I cannot work with SwiftKey: all the time it is going to propose something out of subject, why it proposes all the time my town or my name ?

There could be several reasons for this. Maybe you accidentally gave it some permissions like `Location`/`Contacts`/`Storage` and it fed on your data.
Also, its suggestion mechanism is pretty complex. Its documentation doesn't go into details and technicalities, but it learns and memorizes your input not just from every app, but to SwiftKey, every input field, that is - every text line/row is unique and hence its suggestions are unique for every line of text. It works like this:

Imagine you have a notepad which can only have 2 lines/rows of text. You use your 2-lined notepad as a grocery/shopping to-buy list.

1. This is the first line of your notepad. If you type `apples 2 pcs` in this line with SwiftKey, it will learn that the phrase `apples 2 pcs` belongs to this specific app (notepad) and to this specific line of your notepad, and next time it will suggest it to you if your cursor is currently on the first line of your notepad.
2. I just hit Enter and started a second line. Note that by the word "line" I don't mean the length of the text exactly. I mean that as long as we don't press Enter, our text is considered as a separate and unique input field (no matter the length) with its own unique predictions, belonging only to this line of text. When we hit Enter, what essentially happens is that it inserts a character which declares an end of line (it can be a [Carriage return](https://en.wikipedia.org/wiki/Carriage_return) character alone or along with a [Newline](https://en.wikipedia.org/wiki/Newline)). After hitting Enter, a new line begins and SwiftKey will not suggest you the phrase `apples 2 pcs` here, as SwiftKey sees the second line as separate/unique, and it will have its own predictions. Here, it will suggest you anything you wrote in this specific line. If you write `oranges 4 pcs` in this line, it will learn this and next time you open your notepad and move the cursor to the second line, it will suggest it.

Also, though I'm sure you know this, you can remove a suggestion that you don't want from the prediction bar by long-tapping it. 
Also, you can turn off data sharing in SwiftKey in settings.

That said, your points and preference is 100% valid. To each his own.

**Comment by DwainZwerg** [comment]
Do you know https://github.com/Julow/Unexpected-Keyboard? It's a very good FOSS keyboard.

**Comment by philibe** [comment]
@0-Il :

You're right. Your explanation is good and understandable. :)

But I think in SwiftKey there is no possibility of no suggestion: I have deleted regularly the datas and cache of SwiftKey.

You're right it's my own opinion but  I understand that some other people want suggestions and so one.

But for me it was a very important point  of choice of Openboard: to have a minimalist keyboard.

@DwainZwerg :

I note it. Thanks :)

**Comment by Helium314** [comment]
I decided to piece together some of the open PRs and some more stuff I had started working on a year ago (and never finished because devs wouldn't merge it anyway).
Some parts are not really tested, some settings require reloading the keyboard to work (kill the app, or switch to a different keyboard and back). use at your own risk.
https://github.com/Helium314/openboard/releases/tag/v1.4.5_new_v2

**Comment by ghost** [comment]
@Helium314

This is great. Good job and thanks!

**Comment by bushrang3r** [comment]
Cool! Thanks @Helium314 
Trying it out 🙌

**Comment by Apacelus** [comment]
@Helium314 can you please enable issues in your repo

**Comment by l3u** [comment]
@Helium314 Thanks a lot for your work! I'm just typing this using your package, and finally, I'm rid of that annoying „ß“ shift :-)

Do you plan to fork the project? It would be really, really cool if updates (possibly also calling it another name or such) would be available again on F-Droid.

Sadly, I can't help much … I know a little C++ (mostly in combination with Qt), but when it comes to Java/Kotlin and Android, I'm out …

However, big thanks again!

**Comment by Helium314** [comment]
I enabled issues, but please use them mostly for reporting bugs in the added features. I plan to look into some of the other issues in this repo already, but can't promise anything...

> Do you plan to fork the project? It would be really, really cool if updates (possibly also calling it another name or such) would be available again on F-Droid.

It's an idea, but I currently don't really want to as I fear it will take a lot of time...

**Comment by SynapseScribe** [comment]
this was the best keyboard for me. is it abandoned? if yes, what other open source similar alternatives are out there?

**Comment by ghost** [comment]
@fusionneur

https://github.com/openboard-team/openboard/issues/787#issuecomment-1612040629

**Comment by SynapseScribe** [comment]
could you please add ability to search emojis and also update the list if any missing? also there are cases when word suggestions do not appear for some reason 🚀

**Comment by soshial** [comment]
What about another FOSS keyboard: [Simple Keyboard](https://github.com/rkkr/simple-keyboard)?
1. It's open source
2. it's based on AOSP
3. the main dev is active and accepts PRs.

**Comment by Apacelus** [comment]
> What about another FOSS keyboard: [Simple Keyboard](https://github.com/rkkr/simple-keyboard)?
> 
>     1. It's open source
> 
>     2. it's based on AOSP
> 
>     3. the main dev is active and accepts PRs.

4. No predictive typing...

**Comment by bushrang3r** [comment]
> 4. No predictive typing...
This is the one reason that I didn't change to Simple Keyboard 😟

**Comment by mrx23dot** [comment]
openboard lacks a lot when it comes to email typing :(

- Doesn't auto remember emails
- no selecting text to add to dictionary
- no frequently used phrases

I'm really trying hard not going back to gBoard.

**Comment by ghost** [comment]
@mrx23dot

OpenBoard was forked: https://github.com/Helium314/openboard/releases/latest
It has additional features, try it.
